mono/metadata/class-init.c

```diff
@@ -5044,7 +5044,7 @@ mono_class_setup_methods (MonoClass *klass)
 		int slot = 0;
 		/*Only assign slots to virtual methods as interfaces are allowed to have static methods.*/
 		for (i = 0; i < count; ++i) {
-			if (methods [i]->flags & METHOD_ATTRIBUTE_VIRTUAL)
+			if (!(methods [i]->flags & METHOD_ATTRIBUTE_STATIC))
```

Comment:
Please update the comment above the loop to mention sealed methods.  Otherwise someone might revert this change if they decide to "cleanup" the code.